open_ce/build_tree.py

```diff
@@ -513,45 +496,87 @@ def get_test_feedstocks(self, variant_string):
         """
         return self._test_feedstocks[variant_string]
 
-    def _detect_cycle(self, max_cycles=10):
-        extract_build_tree = [x.build_command_dependencies for x in self.build_commands]
-        cycles = []
-        for start in range(len(self.build_commands)): # Check to see if there are any cycles that start anywhere in the tree.
-            cycles += find_all_cycles(extract_build_tree, start)
-            if len(cycles) >= max_cycles:
-                break
-        if cycles:
-            cycle_print = "\n".join([" -> ".join([self.build_commands[i].recipe
-                                                                    for i in cycle])
-                                                                    for cycle in cycles[:min(max_cycles, len(cycles))]])
-            if len(cycles) > max_cycles:
-                cycle_print += "\nCycles truncated after {}...".format(max_cycles)
-            raise OpenCEError(Error.BUILD_TREE_CYCLE, cycle_print)
-
-
-def find_all_cycles(tree, current=0, seen=None):
-    '''
-    This function performs a depth first search of a tree from current, returning all cycles
-    starting at current.
-    '''
-    if not seen:
-        seen = []
-    current_branch = seen + [current]
-    if len(current_branch) != len(set(current_branch)):
-        return [current_branch]
-    result = []
-    for dependency in tree[current]:
-        next_step = find_all_cycles(tree, dependency, current_branch)
-        if next_step:
-            result += next_step
-    return result
-
-def get_installable_packages(build_commands, external_deps, package_indices=None):
+    def _detect_cycle(self):
```

Comment:
I don't think we are calling `_detect_cycle` anymore.

Reply:
We're not, but I didn't want to get rid of it in case we wanted to use it in the future.

Reply:
OK, then we will need to leave a test in a test for it as well.

Reply:
That makes sense, so it doesn't decay. I'll do this as well.

Reply:
I added cycle detection back, but it only checks packages being built.